server/common/colors.py

```diff
@@ -0,0 +1,228 @@
+    mediumseagreen="#3cb371",
+    mediumslateblue="#7b68ee",
+    mediumspringgreen="#00fa9a",
+    mediumturquoise="#48d1cc",
+    mediumvioletred="#c71585",
+    midnightblue="#191970",
+    mintcream="#f5fffa",
+    mistyrose="#ffe4e1",
+    moccasin="#ffe4b5",
+    navajowhite="#ffdead",
+    navy="#000080",
+    oldlace="#fdf5e6",
+    olive="#808000",
+    olivedrab="#6b8e23",
+    orange="#ffa500",
+    orangered="#ff4500",
+    orchid="#da70d6",
+    palegoldenrod="#eee8aa",
+    palegreen="#98fb98",
+    paleturquoise="#afeeee",
+    palevioletred="#db7093",
+    papayawhip="#ffefd5",
+    peachpuff="#ffdab9",
+    peru="#cd853f",
+    pink="#ffc0cb",
+    plum="#dda0dd",
+    powderblue="#b0e0e6",
+    purple="#800080",
+    rebeccapurple="#663399",
+    red="#ff0000",
+    rosybrown="#bc8f8f",
+    royalblue="#4169e1",
+    saddlebrown="#8b4513",
+    salmon="#fa8072",
+    sandybrown="#f4a460",
+    seagreen="#2e8b57",
+    seashell="#fff5ee",
+    sienna="#a0522d",
+    silver="#c0c0c0",
+    skyblue="#87ceeb",
+    slateblue="#6a5acd",
+    slategray="#708090",
+    slategrey="#708090",
+    snow="#fffafa",
+    springgreen="#00ff7f",
+    steelblue="#4682b4",
+    tan="#d2b48c",
+    teal="#008080",
+    thistle="#d8bfd8",
+    tomato="#ff6347",
+    turquoise="#40e0d0",
+    violet="#ee82ee",
+    wheat="#f5deb3",
+    white="#ffffff",
+    whitesmoke="#f5f5f5",
+    yellow="#ffff00",
+    yellowgreen="#9acd32",
+)
+
+
+class ColorFormatException(Exception):
+    """Raised when color helper functions encounter an unknown color format"""
+
+    pass
+
+
+def convert_color_to_hex_format(unknown):
+    """
+    Try to convert color info to a hex triplet string https://en.wikipedia.org/wiki/Web_colors#Hex_triplet.
+
+    The function accepts for the following formats:
+    - A CSS4 color name, as supported by matplotlib https://matplotlib.org/3.1.0/gallery/color/named_colors.html
+    - RGB tuple/list with values ranging from 0.0 to 1.0, as in [0.5, 0.75, 1.0]
+    - RFB tuple/list with values ranging from 0 to 255, as in [128, 192, 255]
+    - Hex triplet string, as in "#08c0ff"
+
+    :param unknown: color info of unknown format
+    :return: a hex triplet representing that color
+    """
+    if type(unknown) in (list, tuple):
+        if all(0.0 <= ele <= 1.0 for ele in unknown):
+            tup = tuple(int(ele * 255) for ele in unknown)
+        elif all(0 <= ele <= 255 and isinstance(ele, int) for ele in unknown):
+            tup = tuple(unknown)
+        else:
+            raise ColorFormatException("Unknown color format!")
+        return "#%02x%02x%02x" % tup
+    elif isinstance(unknown, str) and unknown in CSS4_NAMED_COLORS:
+        return CSS4_NAMED_COLORS[unknown]
+    elif isinstance(unknown, str) and HEX_COLOR_FORMAT.match(unknown):
+        return unknown.lower()
+    else:
+        raise ColorFormatException("Unknown color format!")
+
+
+def convert_anndata_category_colors_to_cxg_category_colors(data):
```

Comment:
what is the expected behavior on malformed data?  Looks it will throw a variety of exceptions, but not all are caught and processed the same upstream (as written, `convert_color_to_hex_format` can generate a TypeError or a ColorFormatException, maybe more, and only the latter is caught/handled in the REST route).

Worth adding this to the docstring, and then adding test cases for common malformed data.

Reply:
Added documentation and wrapped `convert_color_to_hex_format` logic so that it always returns a `ColorFormatException`